File: vexil_mobile/vexil_app/lib/utils/format.dart
import 'dart:math';

/// 格式化文件大小
String formatSize(int bytes) {
  if (bytes < 1024) return '$bytes B';
  if (bytes < 1024 * 1024) return '${(bytes / 1024).toStringAsFixed(1)} KB';
  if (bytes < 1024 * 1024 * 1024) {
    return '${(bytes / (1024 * 1024)).toStringAsFixed(1)} MB';
  }
  return '${(bytes / (1024 * 1024 * 1024)).toStringAsFixed(1)} GB';
}

/// 格式化速度（bytes/s → 可读字符串）
String formatSpeed(double bytesPerSec) {
  if (bytesPerSec < 1024) return '${bytesPerSec.toInt()} B/s';
  if (bytesPerSec < 1024 * 1024) {
    return '${(bytesPerSec / 1024).toStringAsFixed(1)} KB/s';
  }
  if (bytesPerSec < 1024 * 1024 * 1024) {
    return '${(bytesPerSec / 1024 / 1024).toStringAsFixed(1)} MB/s';
  }
  return '${(bytesPerSec / 1024 / 1024 / 1024).toStringAsFixed(1)} GB/s';
}

/// 格式化预估剩余时间
String formatETA(int seconds) {
  if (seconds <= 0) return '';
  if (seconds < 60) return '${seconds}s';
  if (seconds < 3600) {
    final m = seconds ~/ 60;
    final s = seconds % 60;
    return '${m}m${s}s';
  }
  final h = seconds ~/ 3600;
  final m = (seconds % 3600) ~/ 60;
  return '${h}h${m}m';
}

File: vexil_desktop/internal/gui/format.go
package gui

import (
	"fmt"
	"time"

	"vexil/internal/util"
)

func formatSize(bytes int64) string {
	return util.FormatSize(bytes)
}

func formatSpeed(mbps float64) string {
	if mbps < 1 {
		return fmt.Sprintf("%.1f KB/s", mbps*1024)
	}
	return fmt.Sprintf("%.1f MB/s", mbps)
}

func formatETA(d time.Duration) string {
	return util.FormatETA(d)
}

func errorString(err error) string {
	if err == nil {
		return ""
	}
	return err.Error()
}